<?xml version="1.0"?>
<launch>
  <rosparam file="$(find roberto_hw_interface)/config/controllers.yaml" command="load"/>
  <rosparam file="$(find roberto_hw_interface)/config/joint_limits.yaml" command="load"/>

  <node name="RobertoHardwareInterface" pkg="roberto_hw_interface" type="roberto_hw">
      <param name="drive_kP"    value="0.1"/>
      <param name="drive_kI"    value="0.002"/>
      <param name="drive_kD"    value="5.0"/>
      <param name="drive_kF"    value="0.035"/>
      <param name="drive_ramp"  value="1"/>
  </node>

  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
            /Roberto/joints_update
            /Roberto/auger_controller
            /Roberto/actuator_pos_controller
            /Roberto/bscrew_pos_controller
            /Roberto/left_wheel_controller
            /Roberto/right_wheel_controller"> 
  </node>
  
  <node name="DiffDrive" pkg="roberto_hw_interface" type="diff_drive">
      <param name="totalMod" value="1"/>
      <param name="leftMod" value="1"/>
      <param name="rightMod" value="1"/>
      <param name="turnMod" value="0.5"/>
      <remap from="/outputLeft" to="/Roberto/left_wheel_controller/command"/>
      <remap from="/outputRight" to="/Roberto/right_wheel_controller/command"/>
      <remap from="/input" to="/Drivetrain/cmd_vel"/>
  </node>

  <node name="joystickpython" pkg="roberto_hw_interface" type="moveRobot.py" output="screen">
    <remap from="/bscrew_cmd" to="/Roberto/bscrew_pos_controller/command"/>
    <remap from="/actuator_cmd" to="/Roberto/actuator_pos_controller/command"/>
    <remap from="/auger_cmd" to="/Roberto/auger_controller/command"/>
    <remap from="/drivetrain_cmd" to="/Drivetrain/cmd_vel"/>
    <!--<remap from="/bscrew_pos" to="/bscrew_pos_state"/>-->
  </node>

</launch>